### DISCUSSION TITLE
how to make scala-cli repl behave similarly to sbt console

### DISCUSSION THREAD

**Discussion by ag91** [VirtusLab/scala-cli#2511] [Ideas]
Thanks for the amazing scala-cli!

I like to open a repl to test ideas out and I find it awesome when I can just use the context of the file I am looking at.
Yesterday I discovered that I can do that jn an sbt project with scala-cli through the following steps:
1. add sbt assembly plugin (if it is not already there)
2. append to build sbt
```
assemblyMergeStrategy in assembly := {
  case PathList("META-INF", xs @ _*) => MergeStrategy.discard
  case x => MergeStrategy.first
}
```
3. start scala-cli with --jar option and the uberjar created by the assembly plugin

it requires a few steps, so I was planning to automate this.

Any chance this is an undocumented feature that scala-cli already support? If so, how can I do it in a simpler way before I start automating things?
And if not would it be a feature interesting to anyone else? I could share the script :)

**Comment by SethTisue** [comment]
like this?

```
% echo 'case class C(x: Int)' > C.scala
% scala-cli repl .
Compiling project (Scala 3.3.1, JVM (17))
Compiled project (Scala 3.3.1, JVM (17))
Welcome to Scala 3.3.1 (17.0.9, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                    
scala> C(5)
val res0: C = C(5)
```

**Reply by ag91** [reply]
that is cool indeed!
My context is a bit more complex though: I am in an **sbt project** that comes with external dependencies.
To give a sense: if you run `sbt new http4s/http4s.g8 --branch 1.0` and try to run `scala-cli .` in the source directory with `QuickstartRoutes.scala`, you will see compilation errors because scala-cli didn't build the project and so doesn't know about the dependencies used.

My way around was to make a jar with all deps and feed that to scala-cli, so that I can hack in the repl the contents of the file I need (in my example I can run or redefine `jokeRoutes` because the `QuickstartRoutes.scala` has been packaged in the uberjar).
` `

**Reply by SethTisue** [reply]
Does sbt’s `console` command do what you need?

**Reply by ag91** [reply]
yup, sbt console works fine. 
I was just wondering if I could make scala-cli do the same in a simpler way than creating a uberjar manually.
I am trying to make this a smooth user experience  in Emacs and I would need to integrate sbt console for that while I already integrated scala-cli.

**Reply by SethTisue** [reply]
You could `export fullClasspath` from sbt and then feed that to `scala-cli repl -classpath`.

**Reply by ag91** [reply]
that made the trick, thank you @SethTisue ! Just expanded my little Emacs extension: so cool!
